frontend/app/routes/protected/sin-application/primary-docs-form.tsx

```diff
@@ -0,0 +1,258 @@
+import type { JSX } from 'react';
+import { useState } from 'react';
+
+import type { RouteHandle } from 'react-router';
+
+import { useTranslation } from 'react-i18next';
+
+import type {
+  LocalizedApplicantGender,
+  LocalizedApplicantPrimaryDocumentChoice,
+  LocalizedApplicantStatusInCanadaChoice,
+} from '~/.server/domain/person-case/models';
+import { DatePickerField } from '~/components/date-picker-field';
+import { InputField } from '~/components/input-field';
+import { InputFile } from '~/components/input-file';
+import { InputRadios } from '~/components/input-radios';
+import { InputSelect } from '~/components/input-select';
+import { APPLICANT_PRIMARY_DOCUMENT_CHOICE, APPLICANT_STATUS_IN_CANADA } from '~/domain/constants';
+import type { Errors, PrimaryDocumentData } from '~/routes/protected/person-case/state-machine-models';
+import { getSingleKey } from '~/utils/i18n-utils';
+
+export const handle = {
+  i18nNamespace: ['protected'],
+} as const satisfies RouteHandle;
+
+interface PrimaryDocsFormProps {
+  formValues: Partial<PrimaryDocumentData> | undefined;
+  formErrors: Errors;
+  localizedPrimaryDocumentChoices: LocalizedApplicantPrimaryDocumentChoice[];
+  localizedStatusInCanada: LocalizedApplicantStatusInCanadaChoice[];
+  localizedGenders: LocalizedApplicantGender[];
+}
+
+export default function PrimaryDocsForm({
+  formValues,
+  localizedPrimaryDocumentChoices,
+  localizedStatusInCanada,
+  localizedGenders,
+  formErrors,
+}: PrimaryDocsFormProps) {
```

Comment:
This should probably have a return type of `JSX.Element`

Reply:
I didn't mention it in my initial comment (sorry).. but *all* react components in this file should probably have a return type of `JSX.Element`.